index.bs

```diff
@@ -8972,6 +9018,371 @@ worker=] algorithm:
+
+### Types ### {#module-storage-types}
+
+#### The storage.PartitionKey Type #### {#type-storage-PartitionKey}
+
+[=Remote end definition=] and [=Local end definition=]
+
+<pre class="cddl local-cddl remote-cddl">
+storage.PartitionKey {
+    ? userContext: text,
+    ? sourceOrigin: text,
+    Extensible,
+}
+</pre>
+
+The <code>storage.PartitionKey</code> type represents a [=storage partition key=].
+
+A [=remote end=] has a [=/set=] of <dfn>storage partition key attributes</dfn> which contains zero or more implementation-defined string values.
+
+A [=remote end=] has a [=/map=] of <dfn>default values for storage partition key attributes</dfn> which contains zero or more implementation-defined entries. Each key must be a member of the [=storage partition key attributes=] which the [=remote end=] does not require as input to [=storage.getCookies=], [=storage.setCookie=], and [=storage.deleteCookies=]; the corresponding value is a the value which the [=remote end=] uses when one is not provided.
+
+<div algorithm>
+To <dfn>expand a storage partition spec</dfn> given |partition spec| and |cookie domain|:
+
+1. If |partition spec| is a [=browsing context id=]:
+
+  1. Let |browsing context| be the result of [=trying=] to [=get a browsing context=] given |partition spec|.
+
+  1. Let |partition key| be the [=storage partition key|key=] of |browsing context|'s [=associated storage partition=].
+
+  1. Return [=success=] with data |partition key|.
+
+1. Otherwise, if |partition spec| is null:
+
+  1. Let |partition key| be an empty [=/map=].
+
+1. Otherwise:
+
+  1. Assert: |partition spec| is a [=/map=] with zero or more entries.
+
+  1. Let |partition key| be |partition spec|.
+
+1. For each |name| → |default value| in the [=default values for storage partition key attributes=]:
+
+  1. If |partition key|[|name|] does not [=map/exist=]:
+
+    1. [=map/Set=] |partition key|[|name|] to |default value|.
+
+1. If |partition key|["<code>sourceOrigin</code>"] does not [=map/exist=]:
+
+  1. If |cookie domain| is null, then:
+
+    1. Return [=error=] with [=error code=] [=unknown source origin=].
+
+  1. [=map/Set=] |partition key|["<code>sourceOrigin</code>"] to |cookie domain|.
+
+1. Return [=success=] with data |partition key|.
+
+</div>
+
+<div algorithm>
+To <dfn>get the cookie store</dfn> given |storage partition key|:
+
+1. If a [=cookie store=] |match| exists for |storage partition key|:
+
+  1. Return [=success=] with data |match|.
+
+1. Return [=error=] with [=error code=] [=no such storage partition=].
+
+</div>
+
+<div algorithm>
+To <dfn>match cookie</dfn> given |stored cookie| and |filter|:
+
+1. For each |name| → |value| in |filter|:
+
+  1. If |name| is "<code>value</code>":
+
+    1. Set |value| to [=deserialize protocol bytes=] with |value|.
+
+  1. Let |field name| be the field name corresponding to the JSON key |name| in
+    the [=table for cookie conversion=].
+
+  1. If |stored cookie|[|field name|] does not equal |value|:
```

Comment:
I'm not sure how **`equal`** defined for dictionary, and specifically for `network.BytesValue`, which has `value` and `type` fields.

Reply:
Which operand of "equal" are you referring to? I believe the left-hand side is limited to strings, integers, and booleans (since they come from the stored cookie as described by [WebDriver Classic's table](https://w3c.github.io/webdriver/#dfn-table-for-cookie-conversion)). The corresponding types in the "filter" map a defined similarly, with the exception of the property named "value". I think this is what you had in mind with your reference to `network.BytesValue`, but I believe I handled that case with the earlier step's invocation of the "deserialize protocol bytes" algorithm.

Reply:
you are right, it is addressed. Thanks!